ironfish/src/wallet/wallet.test.ts

```diff
@@ -315,9 +315,21 @@ describe('Accounts', () => {
   })
 
   describe('scanTransactions', () => {
+    it('should not rescan when wallet is not started', async () => {
+      const { node } = nodeTest
+      node.wallet['isStarted'] = false
+
+      const connectSpy = jest.spyOn(node.wallet, 'connectBlock')
+
+      await node.wallet.scanTransactions()
+
+      expect(connectSpy).not.toHaveBeenCalled()
+    })
+
     it('should update head status', async () => {
       // G -> 1 -> 2
       const { node } = nodeTest
+      node.wallet['isStarted'] = true
```

Comment:
This might not be needed depending on what you do in the above test.

Reply:
Yeah, this should be started by default so I don't think you need it.

Reply:
<img width="651" alt="image" src="https://github.com/iron-fish/ironfish/assets/13268167/f4bfa759-ed77-48c6-958b-774996a95777">

After putting some logs in and digging deeper, this test seems to call scanTransactions and the isStarted flag is false in that case. So, I do think we need it. Lmk what i'm missing

Reply:
I missed an assumption on what was changed in the wallet file, but I think @hughy's comment below on the rescan case will overlap with this. I'm not sure if the started should be checked here.

Reply:
I think I addressed @hughy 's comments and added an extra test for the case he described.